### PR TITLE
Implement DMN API endpoints

### PR DESCRIPTION
**Pull request by sergei-maertens**
Related to #1680

**Changes**

- Added API endpoint to obtain a list of enabled/available plugins/engines
- Added API endpoint to get a list of available definitions for a given engine
- Added API endpoint to get a list of available versions for a given engine/definition
- Added API endpoint to retrieve the DMN XML for a given engine/definition

### CODE REVIEW

File: src/openforms/dmn/api/views.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    enum=lazy_enum(get_available_engines),
+)
+
+DEFINITION_PARAMETER = OpenApiParameter(
+    name="definition",
+    type=str,
+    location=OpenApiParameter.QUERY,
+    required=True,
+    description=_("Identifier of the definition to retrieve available versions for."),
+)
```
Comment (joeribekker) [inline src/openforms/dmn/api/views.py:48]: I noticed these 2 query params are required.

For the `/xml` endpoint you include the definiton as path-variable. Why do it as query param on other endpoints? Why is the engine not a path-variable? This feels inconsistent.

* `/api/v1/dmn/decision-definitions/versions` has required `engine` and `definition` **query** params.
* `/api/v1/dmn/decision-definitions/{definition}/xml` has required `engine` **query** param and (obviously) required `definition` **path** param
* `/api/v1/dmn/decision-definitions` has required `engine` **query** params - should it be required?

File: src/openforms/dmn/contrib/camunda/plugin.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            )
+            for result in results
+        ]
+
+    @staticmethod
+    def get_definition_xml(definition_id: str, version: str = "") -> str:
+        with get_client() as client:
+            camunda_id = _get_decision_definition_id(client, definition_id, version)
+            xml_response = client.get(f"decision-definition/{camunda_id}/xml")
+        return xml_response["dmn_xml"]
```
Comment (joeribekker) [inline src/openforms/dmn/contrib/camunda/plugin.py:121]: Should have asked in previous PR but is `check_config` implemented? I thought I saw it but would have expected it here.

Comment (sergei-maertens) [reply src/openforms/dmn/contrib/camunda/plugin.py:121]: ah no, I forgot about it when porting the hackathon code!

Comment (sergei-maertens) [reply src/openforms/dmn/contrib/camunda/plugin.py:121]: Added that in #1683

**Pull request merged**